files/external_node_v2.rb

```diff
@@ -377,15 +382,27 @@ def watch_and_send_facts(parallel)
             upload_facts(certname, req)
           end
 
-          result = enc(certname)
-          cache(certname, result)
+          if ! push_only_current_client_facts
```

Comment:
I'd prefer still setting the `result` variable. Makes it easier to reason about the program. Perhaps a `blank_enc` function that returns the empty content would be useful?